src/nodes.coffee

```diff
@@ -2344,6 +2367,12 @@ exports.ImportClause = class ImportClause extends Base
 
     code
 
+  ast: ->
+    compact flatten [
+      @defaultBinding?.ast()
+      @namedImports?.ast()
+    ]
```

Comment:
@helixbass wrote:

`ImportClause`‘s AST ends up as `specifiers` on `ImportDeclaration` and is expected to be an array of specifier ASTs. `@defaultBinding` is an `ImportDefaultDeclaration` whose AST is a single object. `@namedImports` is either an `ImportNamespaceDeclaration` (whose AST is a single object) or an `ImportSpecifierList` (whose AST is a list of `ImportSpecifier` objects)

So `flatten` is a convenient way to iron that out into the expected shape (a non-nested list of specifier objects)

`compact` combined with the `?.ast()` guards is a way to avoid building up the list manually via conditionals (it’ll get rid of the `undefined`s from when `@defaultBinding` or `@namedImports` doesn’t exist)

Without either you’d end up with something like
```coffee
ast: ->
  ret = []
  if @defaultBinding?
    ret.push @defaultBinding.ast()
  if @namedImports?
    if @namedImports instanceof ImportNamespaceSpecifier
      ret.push @namedImports.ast()
    else
      ret.push @namedImports.ast()...
  ret
```

Reply:
@helixbass wrote:

`ImportClause`‘s AST ends up as `specifiers` on `ImportDeclaration` and is expected to be an array of specifier ASTs. `@defaultBinding` is an `ImportDefaultDeclaration` whose AST is a single object. `@namedImports` is either an `ImportNamespaceDeclaration` (whose AST is a single object) or an `ImportSpecifierList` (whose AST is a list of `ImportSpecifier` objects)

So `flatten` is a convenient way to iron that out into the expected shape (a non-nested list of specifier objects)

`compact` combined with the `?.ast()` guards is a way to avoid building up the list manually via conditionals (it’ll get rid of the `undefined`s from when `@defaultBinding` or `@namedImports` doesn’t exist)

Without either you’d end up with something like
```coffee
ast: ->
  ret = []
  if @defaultBinding?
    ret.push @defaultBinding.ast()
  if @namedImports?
    if @namedImports instanceof ImportNamespaceSpecifier
      ret.push @namedImports.ast()
    else
      ret.push @namedImports.ast()...
  ret
```